### PR TITLE
Add section about data injection

### PR DESCRIPTION
**Pull request by Chr1st1anSears**
This PR adds a new section to our page about creating a custom integration test. The new section describes parameters and labels that RHTAP/App Studio injects into the PipelineRuns of these tests, and how that data is relevant to users.

### CODE REVIEW

**Comment by Chr1st1anSears** [comment]
@jinqi7 here's the new PR. Would you and @dirgim please take a look and see if I've captured all the technical info that was missing? (Heads up: I got rid of the NOTE part, but I included that info at the beginning of the NAMESPACE bullet point.) And @gtrivedi88 would you please give me a peer review for the new section? Thanks.

**Review by dirgim — APPROVED** [review]
This looks good to me, I'll defer to @jinqi7 for the final say on this.

**Review by jinqi7 — APPROVED** [review]

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 1]
```diff
@@ -91,6 +91,23 @@ spec:
 . Add your new custom test as an integration test in {ProductName}.
 .. For additional instructions on adding an integration test, see this document: xref:how-to-guides/testing_applications/proc_adding_an_integration_test.adoc[Adding an integration test].
 
+.Data injected into the PipelineRun of the integration test
+
+When you create a custom integration test, {ProductName} automatically adds certain parameters and labels to the PipelineRun of the integration test. This section explains what those parameters and labels are, and how they can help you.
+
+Parameters:
+
+* `SNAPSHOT`: contains the xref:../../glossary/index.adoc#_snapshot[snapshot] of the whole application as a JSON string. This JSON string provides useful information about the test, such as which components {ProductName} is testing, and what git repository and commit {ProductName} is using to build those components. Click link:https://github.com/redhat-appstudio/integration-examples/blob/main/examples/snapshot_json_string_example[here] to see an example of a snapshot JSON string.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:100]:
```suggestion
* *`SNAPSHOT`*: contains the xref:../../glossary/index.adoc#_snapshot[snapshot] of the whole application as a JSON string. This JSON string provides useful information about the test, such as which components {ProductName} is testing, and what git repository and commit {ProductName} is using to build those components. For information on snapshot JSON string, see link:https://github.com/redhat-appstudio/integration-examples/blob/main/examples/snapshot_json_string_example[snapshot json string example]
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 2]
```diff
@@ -91,6 +91,23 @@ spec:
...
+* `NAMESPACE`: If your integration test specifies that {ProductName} should deploy the snapshot to an ephemeral environment during testing, then {ProductName} adds this parameter to indicate which namespace it uses for that deployment. This parameter can help you query Kubernetes resources that are deployed in the test environment, such as routes, deployments, and pods.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:101]:
```suggestion
* *`NAMESPACE`*: If your integration test specifies that {ProductName} should deploy a snapshot to an ephemeral environment during testing, then {ProductName} adds the `namespace` parameter to indicate which namespace it uses for that deployment. This parameter can help you query Kubernetes resources that are deployed in the test environment, such as routes, deployments, and pods. 
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 3]
```diff
@@ -91,6 +91,23 @@ spec:
...
+
+Parameters:
+
+* `SNAPSHOT`: contains the xref:../../glossary/index.adoc#_snapshot[snapshot] of the whole application as a JSON string. This JSON string provides useful information about the test, such as which components {ProductName} is testing, and what git repository and commit {ProductName} is using to build those components. Click link:https://github.com/redhat-appstudio/integration-examples/blob/main/examples/snapshot_json_string_example[here] to see an example of a snapshot JSON string.
+* `NAMESPACE`: If your integration test specifies that {ProductName} should deploy the snapshot to an ephemeral environment during testing, then {ProductName} adds this parameter to indicate which namespace it uses for that deployment. This parameter can help you query Kubernetes resources that are deployed in the test environment, such as routes, deployments, and pods. 
+
+Labels:
+
+* `appstudio.openshift.io/application`: contains the name of the application.
+* `appstudio.openshift.io/component`: contains the name of the component.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:106]:
```suggestion
* *`appstudio.openshift.io/component`*: contains the name of the component.
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 4]
```diff
@@ -91,6 +91,23 @@ spec:
...
+
+Parameters:
+
+* `SNAPSHOT`: contains the xref:../../glossary/index.adoc#_snapshot[snapshot] of the whole application as a JSON string. This JSON string provides useful information about the test, such as which components {ProductName} is testing, and what git repository and commit {ProductName} is using to build those components. Click link:https://github.com/redhat-appstudio/integration-examples/blob/main/examples/snapshot_json_string_example[here] to see an example of a snapshot JSON string.
+* `NAMESPACE`: If your integration test specifies that {ProductName} should deploy the snapshot to an ephemeral environment during testing, then {ProductName} adds this parameter to indicate which namespace it uses for that deployment. This parameter can help you query Kubernetes resources that are deployed in the test environment, such as routes, deployments, and pods. 
+
+Labels:
+
+* `appstudio.openshift.io/application`: contains the name of the application.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:105]:
```suggestion
* *`appstudio.openshift.io/application`*: contains the name of the application.
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 5]
```diff
@@ -91,6 +91,23 @@ spec:
...
+* `appstudio.openshift.io/component`: contains the name of the component.
+* `appstudio.openshift.io/snapshot`: contains the name of the snapshot.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:107]:
```suggestion
* *`appstudio.openshift.io/snapshot`*: contains the name of the snapshot.
```

**Comment by gtrivedi88** [comment]
@Chr1st1anSears Thank you so much for your hard work on this PR. I've added some suggestions for consistency and other things. If you have any questions, please let me know.

**Pull request merged**